dot/parachain/pvf/host.go

```diff
@@ -0,0 +1,148 @@
+package pvf
+
+import (
+	"fmt"
+	"sync"
+
+	parachainruntime "github.com/ChainSafe/gossamer/dot/parachain/runtime"
+	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+var logger = log.NewFromGlobal(log.AddContext("pkg", "pvf"), log.SetLevel(log.Debug))
+
+type ValidationHost struct {
+	wg     sync.WaitGroup
+	stopCh chan struct{}
+
+	workerPool *validationWorkerPool
+}
+
+func (v *ValidationHost) Start() {
+	v.wg.Add(1)
```

Comment:
We must add `wg.Add()`, before calling the goroutine we want to wait on.
Then we pass `wg` inside that goroutine. 
Then we call `wg.Done` at the end of our goroutine.

There might be some good blogs on the topic, but posting one from my search results here https://blog.logrocket.com/concurrency-patterns-golang-waitgroups-goroutines/